### PR TITLE
[optim] More cleanup and reorg of test_optim.py

### PR DESCRIPTION
**Pull request by janeyx99**
Stack from [ghstack](https://github.com/ezyang/ghstack) (oldest at bottom):
* __->__ #100917
* #100899
* #100898

### CODE REVIEW

File: test/test_optim.py [hunk 1]
```diff
@@ -79,28 +79,29 @@ def _test_rosenbrock_sparse(
     ):
         if scheduler_constructors is None:
             scheduler_constructors = []
-        params_t = torch.tensor([1.5, 1.5])
+        # For rosenbrock tests, it is mandated that the param is a tensor with 2 numbers
```
Comment (awgu) [inline test/test_optim.py:82]: nit: Maybe add an explicit `assert` to `rosenbrock()` and `drosenbrock()` to not need the 3 comments (here and before the function definitions)?
```
assert tensor.size() == torch.Size([2]), (
    f"Requires tensor with 2 scalars but got {tensor.size()}
)
```

**Review by albanD — COMMENTED** [review]
Sounds good!
I agree with Andrew on the assert side

File: test/test_optim.py (hunk 1 above)
Comment (janeyx99) [reply test/test_optim.py:82]: agreed! thanks! 

Am still going to leave the last comment just to stop anyone from experimenting with the param length

**Review by albanD — APPROVED** [review]

**Comment by janeyx99** [comment]
:o when did drci get added as a command

**Comment by malfet** [comment]
> :o when did drci get added as a command

Last Novemeber, see https://github.com/pytorch/test-infra/pull/1064

**Comment by janeyx99** [comment]
oh wow i must have missed the announcement

**Comment by albanD** [comment]
Doesn't look cli-y :thinking:

**Pull request closed without merging**